### PR TITLE
Only list changed plan settings in summary

### PR DESCRIPTION
**Pull request by cole-h**
##### Checklist

- [ ] Formatted with `cargo fmt`
- [ ] Built with `nix build`
- [ ] Ran flake checks with `nix flake check`
- [ ] Added or updated relevant tests (leave unchecked if not applicable)
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] Linked to related issues (leave unchecked if not applicable)

##### Validating with `install.determinate.systems`

If a maintainer has added the `upload to s3` label to this PR, it will become available for installation via `install.determinate.systems`:

```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/pr/$PR_NUMBER | sh -s -- install
```

---

There are two possible interfaces -- the one provided in 6d735fc03bfbe7c246e4b3ca2b35fd028f49eee7 (current) and the one provided in ee7d24c968750762036a33649e40e9c4afb581b0 (old).

idk which one I prefer, but they're both pretty Not Ideal.

### CODE REVIEW

**Comment by cole-h** [comment]
Before: 
image

After: 
image

Bonus:

using completely default settings:
image

changing/providing one setting:
image

File: src/settings.rs [hunk 1]
```diff
@@ -548,6 +548,8 @@ pub enum InstallSettingsError {
     ),
     #[error("No supported init system found")]
     InitNotSupported,
+    #[error("Could not find the disk `/` resides on")]
```
Comment (cole-h) [inline src/settings.rs:551]: I intentionally didn't make this `target_os = "macos"` even though that's the only place it's used. I feel like it'd be kinda bad to guard this single error in a downstream crate with `target_os` e.g. if they decide to `match` on it at  any point.

File: src/planner/mod.rs [hunk 2]
```diff
@@ -110,21 +122,26 @@ pub trait Planner: std::fmt::Debug + Send + Sync + dyn_clone::DynClone {
     async fn plan(&self) -> Result<Vec<StatefulAction<Box<dyn Action>>>, PlannerError>;
     /// The settings being used by the planner
     fn settings(&self) -> Result<HashMap<String, serde_json::Value>, InstallSettingsError>;
+    /// The default settings being used by the planner
+    async fn default_settings(
```
Comment (Hoverbear) [inline src/planner/mod.rs:126]: This feels like rule breaking and I think both of us mentioned we didn't love the API...

How would you feel about making `configured_settings` something the various planners have to implement, and removing this implementation requirement? Then we can remove the generic `configured_settings` implementation and each of them can implement this:


```rust
    async fn configured_settings(
        &self,
    ) -> Result<HashMap<String, serde_json::Value>, PlannerError> {
        let default = Self::default().await?.settings()?;
        let configured = self.settings()?;

        let mut settings: HashMap<String, serde_json::Value> = HashMap::new();
        for (key, value) in configured.iter() {
            if default.get(key) != Some(value) {
                settings.insert(key.clone(), value.clone());
            }
        }

        Ok(settings)
    }
```

It seems to work for me locally.

Comment (cole-h) [reply src/planner/mod.rs:126]: (Not quite sure what you mean by "rule breaking")

Yeah, that's what I had in ee7d24c968750762036a33649e40e9c4afb581b0, so I think we can just go back to it.

Comment (Hoverbear) [reply src/planner/mod.rs]: Hmm, so the reason we added `default_setting(&self)` is so we could call `Self::default()` yes? Which we could not do due to object safety rules. So we're essentially skirting object safey by forcing the planner to re-implement `Self::default()` without referring to `Self`. It feels like rule breaking. :)

**Review by Hoverbear — APPROVED** [review]

**Pull request merged**